frame/staking/src/lib.rs

```diff
@@ -684,40 +718,76 @@ decl_storage! {
 			linked_map hasher(blake2_256) T::AccountId => ValidatorPrefs;
 
 		/// The map from nominator stash key to the set of stash keys of all validators to nominate.
-		///
-		/// NOTE: is private so that we can ensure upgraded before all typical accesses.
-		/// Direct storage APIs can still bypass this protection.
-		Nominators get(fn nominators):
+		pub Nominators get(fn nominators):
 			linked_map hasher(blake2_256) T::AccountId => Option<Nominations<T::AccountId>>;
 
-		/// Nominators for a particular account that is in action right now. You can't iterate
-		/// through validators here, but you can find them in the Session module.
+		/// The current era index.
 		///
-		/// This is keyed by the stash account.
-		pub Stakers get(fn stakers):
-			map hasher(blake2_256) T::AccountId => Exposure<T::AccountId, BalanceOf<T>>;
+		/// This is the latest planned era, depending on how session module queues the validator
+		/// set, it might be active or not.
+		pub CurrentEra get(fn current_era): Option<EraIndex>;
```

Comment:
Why is this an option? When it is `None`?

Reply:
as discussed on riot:

it is None before the first era is planned, i.e. before the first call for new_era from SessionManager

I wanted to make the least assumption possible. Now staking can be added into a runtime and the first era is planned whenever the session want to start to use it.
If I put zero here then the initial validator set must be provided with a runtime upgrade in case staking is added into the runtime or it start with an empty validator set 🤔 (with genesis things are configured and fine) Anyway the logic looked weird, the first era is planned when it is first planned by session make more sense to me.